src/gennylib/include/gennylib/context.hpp

```diff
@@ -112,35 +112,25 @@ class WorkloadContext : public v1::ConfigNode {
     // no copy or move
     WorkloadContext(WorkloadContext&) = delete;
     void operator=(WorkloadContext&) = delete;
-    WorkloadContext(WorkloadContext&&) = default;
+    WorkloadContext(WorkloadContext&&) = delete;
     void operator=(WorkloadContext&&) = delete;
 
     /**
      * @return all the actors produced. This should only be called by workload drivers.
      */
-    constexpr const ActorVector& actors() const {
-        return _actors;
-    }
+    const ActorVector& actors() const;
```

Comment:
Hm, what's the benefit of putting this in the CPP file? I think in general people should be encouraged to `inline` and `constexpr` things on contexts since they could be performance critical (e.g. interactions with ShareableState). This particular function is probably unlikely to matter, but I'm less sure about some one liners below on the actor and phase contexts.

FWIW The Google style guide says it's OK to inline small functions. https://google.github.io/styleguide/cppguide.html#Inline_Functions

Reply:
Okay good call - reverted most of these moves to the `.cpp`